cl_main.c

```diff
@@ -3112,7 +3112,7 @@ void CL_Init (void)
 
 		// for QW connections
 		Cvar_RegisterVariable(&qport);
-		Cvar_SetValueQuick(&qport, (rand() * RAND_MAX + rand()) & 0xffff);
+		Cvar_SetValueQuick(&qport, (rand() + 1024) & 0xffff);
```

Comment:
I think it was written that way because on windows RAND_MAX is only 32767 so the max port number there would be 33791 with this change.

Reply:
I'll change it to reflect that and not cause signed overflow, thanks.

Reply:
In the updated version the `+ 1024` happens last, but it needs to be before the `& 0xffff` to ensure we don't exceed 65535.

Reply:
On closer inspection `qport` isn't a network port, DP never opens a listening socket on it.  It's an identifier which is networked to the QW server, apparently to work around problems in ancient platforms.  So I amended the commit to omit the +1024 (and squashed it into the previous commit to reduce noise).